File: 240923/Run Length 인코딩/run-length-encoding.cpp
#include <iostream>
using namespace std;

int main() {
    // 여기에 코드를 작성해주세요.
    string word;
    cin >> word;
    string ans[word.length()] = {};
    int count = 0;
    char check;
    for (int i = 0; i < word.length()-1; i ++) {
        if (word[i] == word[i+1] && i != word.length()-2){
            check = word[i];
            count ++;
        }
        else if (word[i] != word[i+1]) {
            ans.push_back(check);
            check = word[i+1];
            ans.push_back(count +1);
            count = 0;
        } 
        else {
            check = word[i];
            count += 1;
            ans.push_back(check);
            ans.push_back(count +1);
        }
    }
    return 0;
}

File: 240923/Run Length 인코딩/run-length-encoding.py
word = input()
ans = []
count = 0
for i in range(len(word)-1):
    if (word[i] == word[i+1]) and (i != len(word)- 2):
        check = word[i]
        count += 1

    elif  word[i] !=word[i+1] :
        ans.append(check)
        check = word[i+1]
        ans.append(count+1)
        count = 0
    else :
        check = word[i]
        count += 1
        ans.append(check)
        ans.append(count +1)

total = 0
for i in range(len(ans)):
    if len(str(ans[i])) == 1:
        total += 1;
    elif len(str(ans[i])) == 2:
        total += 2;
    else:
        total += 3;

print(total)

for i in range(len(ans)):
    print(ans[i], end = '')